assignments/prototypes.js

```diff
@@ -8,101 +8,161 @@
   Each constructor function has unique properites and methods that are defined in their block comments below:
 */
   
-/*
-  === GameObject ===
-  * createdAt
-  * dimensions
-  * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
-*/
+CharacterStats.prototype = Object.create(GameObject.prototype);
+Humanoid.prototype = Object.create(CharacterStats.prototype);
+Char.prototype = Object.create(Humanoid.prototype);
 
-/*
-  === CharacterStats ===
-  * hp
-  * name
-  * takeDamage() // prototype method -> returns the string '<object name> took damage.'
-  * should inherit destroy() from GameObject's prototype
-*/
+function GameObject(gameTrait) {
+  this.createdAt = gameTrait.createdAt;
+  this.dimensions = gameTrait.dimensions;
+}
 
-/*
-  === Humanoid ===
-  * faction
-  * weapons
-  * language
-  * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
-  * should inherit destroy() from GameObject through CharacterStats
-  * should inherit takeDamage() from CharacterStats
-*/
+GameObject.prototype.destroy = function() {
+  return `${this.name} was removed from the game.`;
+}
+
+function CharacterStats(statsTrait) {
+  this.hp = statsTrait.hp;
+  this.name = statsTrait.name;
+  GameObject.call(this, statsTrait);
```

Comment:
Ideally, you would want to do the call first and then set the attributes. This is to avoid any unintended consequences in edge cases.